src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs

```diff
@@ -159,23 +167,26 @@ private void SessionEnded_Handler(object sender, SessionEndEventArgs args)
 
             if (this.processDumpEnabled)
             {
-                try
+                if (this.testStartCount > this.testEndCount || this.alwaysCollectProcessDump)
```

Comment:
What if process throws before test cases start?

Reply:
We have an option to collect dump no matter the number of test cases. This check is just the default behaviour.